verticles/graphql/src/main/java/com/gentics/mesh/graphql/type/AbstractTypeProvider.java

```diff
@@ -726,8 +729,12 @@ public <T> Pair<Predicate<T>,Optional<FilterOperation<?>>> parseFilters(DataFetc
 	 * @param ignorePaging
 	 * @return
 	 */
-	protected DynamicStreamPageImpl<NodeContent> applyNodeFilter(Stream<? extends NodeContent> stream, PagingParameters pagingInfo, Predicate<NodeContent> javaFilter, boolean ignorePaging) {
-		return new DynamicStreamPageImpl<>(stream, pagingInfo, javaFilter, ignorePaging);
+	@SuppressWarnings("unchecked")
+	protected Page<NodeContent> applyNodeFilter(Stream<? extends NodeContent> stream, PagingParameters pagingInfo, Predicate<NodeContent> javaFilter, boolean ignorePaging, Optional<Supplier<Long>> maybeCountSupplier) {
```

Comment:
Main change: if the stream is already (natively) paged, we should not count a total of its elements, but rather use a separate totals call on demand.